### ISSUE TITLE
iOS pushed screen never shows up and freeze the app

### ISSUE THREAD

**Issue by psyzz** [wix/react-native-navigation#428]
Hello,

I encountered a bug on iOS when pushing a new screen.
Basically i'm clicking on an item, calling this.props.navigator.push( {screen: 'app.myscreen'} )
and sometimes the screen never open.
To make the app "clickable" again i need to swipe back (like if i wanted to close the pushed screen) and the screen appears.

---
### Environment
* React Native Navigation version: ^2.0.0-experimental.136
* React Native version: 0.35.0
* Platform(s): iOS
* Device info: Simulator/Device - Debug/Release

**Comment by arshavinho** [comment]
By testing another of my app made with RNN I just realized that this bug is also present and appears quite often

**Comment by guyca** [comment]
@arshavinho We weren't able to reproduce this bug and no other user reported a bug with pushing screens. If you upload a project where this bug reproduces to GitHub, I'm sure someone will be glad to take a look at it. It's probably an issue in your app's logic. I doubt it's related to RNN. 👍

**Comment by arshavinho** [comment]
ok i will try to create a project to reproduces the bug

**Comment by arshavinho** [comment]
After a discussion with @guyca i was able to find this maybe related issue:
https://github.com/wix/react-native-navigation/issues/245

i tried to add the piece of code from littlesome post but the problem was still there and i couldnt swipe back to close the non-open screen. Here a new gif that shows a better view of the issue:
![rnn-bug-opening-screen-plateculture-gif](https://cloud.githubusercontent.com/assets/1016441/20278239/32ec318c-aaa4-11e6-8dc4-217e28b8f960.gif)

To solve my issue i had to comment in RCCViewControler.m this:
`//Bug fix: in case there is a interactivePopGestureRecognizer, it prevents react-native from getting touch events on the left screen area that the gesture handles
    //overriding the delegate of the gesture prevents this from happening while keeping the gesture intact (another option was to disable it completely by demand)
    self.originalInteractivePopGestureDelegate = nil;
    if (self.navigationController != nil && self.navigationController.interactivePopGestureRecognizer != nil)
    {
      id <UIGestureRecognizerDelegate> interactivePopGestureRecognizer = self.navigationController.interactivePopGestureRecognizer.delegate;
      if (interactivePopGestureRecognizer != nil)
      {
        self.originalInteractivePopGestureDelegate = interactivePopGestureRecognizer;
        self.navigationController.interactivePopGestureRecognizer.delegate = self;
      }
    }
`

to:

`    //Bug fix: in case there is a interactivePopGestureRecognizer, it prevents react-native from getting touch events on the left screen area that the gesture handles
    //overriding the delegate of the gesture prevents this from happening while keeping the gesture intact (another option was to disable it completely by demand)
    self.originalInteractivePopGestureDelegate = nil;
//    if (self.navigationController != nil && self.navigationController.interactivePopGestureRecognizer != nil)
//    {
//      id <UIGestureRecognizerDelegate> interactivePopGestureRecognizer = self.navigationController.interactivePopGestureRecognizer.delegate;
//      if (interactivePopGestureRecognizer != nil)
//      {
//        self.originalInteractivePopGestureDelegate = interactivePopGestureRecognizer;
//        self.navigationController.interactivePopGestureRecognizer.delegate = self;
//      }
//    }
`

So basically the slide to right is now disabled in my App

**Comment by kompot** [comment]
I do observe same behaviour as on @arshavinho 's gif. Can't reliably reproduce it but do see it quite often.

**Comment by arshavinho** [comment]
To reproduce the behaviour you can start by clicking outside of the screen and swipe to the right that will fire the onPress event with the issue.

Sometimes the behaviour occurs also when clicking anywhere on a TouchableOpacity ou TouchableHighlight but i never success to reproduce it 100% a of the time

**Comment by Antonides** [comment]
#341 same problem

**Comment by micchyboy1023** [comment]
I have a same problem.
@arshavinho answer solved the problem!

**Comment by SudoPlz** [comment]
I've got the same problem!!

**Comment by grundmanise** [comment]
@guyca  Same problem when running /example app

**Comment by grundmanise** [comment]
#645

**Comment by SudoPlz** [comment]
Ok I'm not 100000% sure that's related, but after commenting out the code above in `RCCViewControler.m` I no longer get an app freeze after swiping, but I do have a few cases on which a `TouchableOpacity` or a `TouchableHighlight` are being tapped (I know that because their bg color changes) but the onPress event never fires.

**Comment by indivisable** [comment]
+1 Same issue here, glad to see there is a temporary fix but this is a real issue so I hope it gets into official

**Comment by indivisable** [comment]
Update: I still ran into issues when I used 'popToRoot' command but only when animation was enabled.

**Comment by grundmanise** [comment]
@indivisable have you tried littlesome's solution https://github.com/wix/react-native-navigation/issues/245?

**Comment by grundmanise** [comment]
#245 freezes the app for me.
